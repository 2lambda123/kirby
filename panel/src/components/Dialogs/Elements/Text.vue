<template>
	<k-text v-if="text" :html="text" />
	<k-box v-else theme="info">This dialog does not define any text</k-box>
</template>

<script>
export const props = {
	props: {
		text: String
	}
};

export default {
	mixins: [props]
};
</script>
