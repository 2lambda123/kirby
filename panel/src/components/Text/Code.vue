<template>
	<!-- @slot Code content -->
	<pre class="k-code"><code><slot /></code></pre>
</template>

<script>
/**
 * @public
 */
export default {}
</script>

<style>
:root {
	--code-color-back: var(--color-black);
	--code-color-icon: var(--color-gray-500);
	--code-color-text: var(--color-gray-200, white);
	--code-font-family: var(--font-mono);
	--code-font-size: 1em;
}
.k-code,
.k-text pre {
	position: relative;
	display: block;
	font-family: var(--code-font-family);
	font-size: var(--code-font-size);
	max-width: 100%;
	padding: 0.75rem;
	border-radius: var(--rounded, 0.5rem);
	background: var(--code-color-back);
	color: var(--code-color-text);
	white-space: nowrap;
	overflow-y: hidden;
	overflow-x: auto;
	line-height: 1.5;
	tab-size: 2;
}
.k-code:not(code),
.k-text pre {
	white-space: pre-wrap;
}
.k-code::before {
	position: absolute;
	content: attr(data-language);
	inset-block-start: 0;
	inset-inline-end: 0;
	padding: 0.5rem 0.5rem 0.25rem 0.25rem;
	font-size: calc(0.75 * var(--text-xs));
	background: var(--code-color-back);
	border-radius: var(--rounded, 0.5rem);
}
</style>
