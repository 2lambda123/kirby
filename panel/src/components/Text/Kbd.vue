<style>
:root {
	--kbd-color-back: var(--color-gray-300);
}

kbd {
	display: inline-flex;
	padding-inline: var(--spacing-1);
	background: var(--kbd-color-back);
	border-radius: var(--rounded);
	font-family: inherit;
}
</style>
