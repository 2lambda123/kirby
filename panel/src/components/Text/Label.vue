<template>
	<component
		:is="element"
		:for="input"
		:class="'k-' + type + '-label'"
		:data-invalid="invalid"
		class="k-label"
	>
		<k-link v-if="link" :to="link">
			<slot />
		</k-link>
		<slot v-else />
		<abbr v-if="invalid" :title="$t(type + '.invalid')">&times;</abbr>
		<abbr v-else-if="required" :title="$t(type + '.required')">âœ¶</abbr>
	</component>
</template>

<script>
export default {
	props: {
		input: {
			type: [String, Number]
		},
		invalid: {
			type: Boolean
		},
		link: {
			type: String
		},
		required: {
			default: false,
			type: Boolean
		},
		tag: {
			default: "label",
			type: String
		},
		type: {
			default: "field",
			type: String
		}
	},
	computed: {
		element() {
			return this.type === "section" ? "h2" : "label";
		}
	}
};
</script>

<style>
.k-label {
	position: relative;
	display: flex;
	align-items: center;
	height: var(--height-xs);
	overflow: clip;
	text-overflow: ellipsis;
	white-space: nowrap;
	font-weight: var(--font-semi);
}
[aria-disabled] .k-label {
	opacity: var(--opacity-disabled);
	cursor: not-allowed;
}

/** Required and invalid sign **/
.k-label abbr {
	color: var(--color-gray-500);
	margin-inline-start: 0.375rem;
}

.k-label[data-invalid] abbr {
	color: var(--color-red-700);
}
</style>
