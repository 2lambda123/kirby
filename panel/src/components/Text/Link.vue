<style>
.k-text :where(.k-link, a) {
	color: var(--link-color);
	text-decoration: underline;
	text-underline-offset: var(--link-underline-offset);
	border-radius: var(--rounded-xs);
	outline-offset: 2px;
}
</style>
