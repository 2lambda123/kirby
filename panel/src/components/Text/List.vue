<style>
:root {
	--link-color: var(--color-blue-800);
	--link-underline-offset: 2px;
}

.k-text :where(.k-link, a) {
	color: var(--link-color);
	text-decoration: underline;
	text-underline-offset: var(--link-underline-offset);
	border-radius: var(--rounded-xs);
	outline-offset: 2px;
}
</style>
