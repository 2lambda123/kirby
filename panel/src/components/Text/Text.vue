<template>
	<!-- eslint-disable-next-line vue/no-v-html -->
	<div v-if="html" v-bind="attrs" v-html="html"></div>
	<div v-else v-bind="attrs">
		<!-- @slot Text content -->
		<slot />
	</div>
</template>

<script>
/**
 * The Text component is a container for
 * all multi-line text with additional formats.
 * @public
 *
 * @example <k-text>
  <b>Lorem</b> <a href="#">ipsum</a> <i>dolor</i> â€¦
</k-text>
 */
export default {
	props: {
		/**
		 * Changes the text alignment
		 * @values start, center, end
		 */
		align: String,
		/**
		 * HTML content to render instead
		 * of the default slot
		 */
		html: String,
		/**
		 * Font size of the text
		 * @values tiny, small, medium, large, huge
		 */
		size: String,
		/**
		 * Visual appearance of the text
		 * @values help
		 * @deprecated Use `k-help` class instead
		 * @todo Remove in 5.0
		 */
		theme: String
	},
	computed: {
		attrs() {
			return {
				class: "k-text",
				"data-align": this.align,
				"data-size": this.size,
				"data-theme": this.theme
			};
		}
	}
};
</script>

<style>
:root {
	--text-font-size: 1em;
	--text-line-height: 1.375;
}

.k-text {
	font-size: var(--text-font-size);
	line-height: var(--text-line-height);
}

.k-text > * + * {
	margin-block-start: calc(var(--text-line-height) * 1em);
}

.k-text > * {
	max-width: 60ch;
}

.k-text ol,
.k-text ul {
	margin-inline-start: 2em;
}

.k-text ol {
	list-style: numeric;
}

.k-text ul {
	list-style: disc;
}

.k-text a {
	text-decoration: underline;
}

.k-text > * + h6 {
	margin-block-start: calc(var(--text-line-height) * 1.5em);
}

.k-text[data-size="tiny"] {
	--text-font-size: var(--text-xs);
}
.k-text[data-size="small"] {
	--text-font-size: var(--text-sm);
}
.k-text[data-size="medium"] {
	--text-font-size: var(--text-md);
}
.k-text[data-size="large"] {
	--text-font-size: var(--text-xl);
}
.k-text[data-align] {
	text-align: var(--align);
}

/** Code */
.h1 code,
.k-text h1 code,
.h2 code,
.k-text h2 code,
.h3 code,
.k-text h3 code,
.h4 code,
.k-text h4 code,
.h5 code,
.k-text h5 code,
.h6 code,
.k-text h6 code {
	font-family: var(--font-mono, monospace);
	font-size: 0.925em;
	font-weight: var(--font-normal);
}

.k-text iframe {
	width: 100%;
	aspect-ratio: 16/9;
}

/** HR **/
.hr,
.k-text hr {
	background: var(--color-border);
	height: 1px;
}

/** Links **/
:root {
	--link-color: var(--color-blue-800);
	--link-underline-offset: 2px;
}

.k-text :where(.k-link, a) {
	color: var(--link-color);
	text-decoration: underline;
	text-underline-offset: var(--link-underline-offset);
	border-radius: var(--rounded-xs);
	outline-offset: 2px;
}

/** Help */
.k-help {
	color: var(--color-text-dimmed);
	font-size: var(--text-xs);
	line-height: 1.25;
}
</style>
