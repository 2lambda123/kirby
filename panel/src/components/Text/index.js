import Headline from "./Headline.vue";
import Text from "./Text.vue";

export default {
	install(app) {
		app.component("k-headline", Headline);
		app.component("k-text", Text);
	}
};
