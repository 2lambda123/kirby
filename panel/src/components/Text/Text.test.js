import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import Text from "./Text.vue";

describe("Text.vue", () => {
	it("has default slot", async () => {
		const wrapper = mount(Text, {
			slots: {
				default: "Foo"
			}
		});

		expect(wrapper.text()).toBe("Foo");
	});

	it("has attributes", async () => {
		const wrapper = mount(Text, {
			propsData: {
				align: "right",
				size: "small"
			}
		});

		expect(wrapper.attributes("data-align")).toBe("right");
		expect(wrapper.attributes("data-size")).toBe("small");
	});

	it("has CSS selector", async () => {
		const wrapper = mount(Text);
		expect(wrapper.classes()).toContain("k-text");
	});

	it("matches snapshot", async () => {
		const wrapper = mount(Text, {
			propsData: {
				align: "right",
				size: "large"
			},
			slots: {
				default: "This is some text"
			}
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});
