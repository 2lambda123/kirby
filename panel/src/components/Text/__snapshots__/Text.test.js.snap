// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text.vue > matches snapshot 1`] = `
<div
  class="k-text"
  data-align="right"
  data-size="large"
  data-theme="help"
>
  This is some text
</div>
`;
